### PR TITLE
Ins 1024 chemical treatment bug

### PR DESCRIPTION
**Pull request by ghost**
- The tabbing issue in
   - herbicide dropdown
   - species treated section
- Validations in the base-form component
- UI fixes
- A little bit of clean-up & refactoring

### CODE REVIEW

File: app/lucy/src/app/directives/click-away/click-away.directive.spec.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { ClickAwayDirective } from './click-away.directive';
 
 describe('ClickAwayDirective', () => {
   it('should create an instance', () => {
-    const directive = new ClickAwayDirective();
+    const directive = new ClickAwayDirective(undefined);
```
Comment (fwpushan) [inline app/lucy/src/app/directives/click-away/click-away.directive.spec.ts]: why explicitly setting undefined

File: app/lucy/src/app/directives/element-focus/element-focus.directive.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { Directive, ElementRef, Input, OnInit } from '@angular/core';
+
+@Directive({
+  selector: '[appElementFocus]'
+})
+export class ElementFocusDirective implements OnInit {
```
Comment (fwpushan) [inline app/lucy/src/app/directives/element-focus/element-focus.directive.ts]: Add Documentation

File: app/lucy/src/app/components/Routes/herbicide-application/herbicide-application.component.ts [hunk 3]
```diff
@@ -130,6 +130,13 @@ export class HerbicideApplicationComponent implements OnInit, AfterViewInit {
       return this._tankMixes;
   }
 
+  get copyTankMixes(): HerbicideTankMix[] {
```
Comment (fwpushan) [inline app/lucy/src/app/components/Routes/herbicide-application/herbicide-application.component.ts]: Method doc

File: app/lucy/src/app/directives/click-away/click-away.directive.spec.ts (hunk 1 above)
Comment (rajfreshworks) [reply app/lucy/src/app/directives/click-away/click-away.directive.spec.ts]: I forgot to change this one actually, will do that 👍

File: app/lucy/src/app/directives/element-focus/element-focus.directive.ts (hunk 2 above)
Comment (rajfreshworks) [reply app/lucy/src/app/directives/element-focus/element-focus.directive.ts]: As `comments` or in the actual `README`?

File: app/lucy/src/app/components/Routes/herbicide-application/herbicide-application.component.ts (hunk 3 above)
Comment (rajfreshworks) [reply app/lucy/src/app/components/Routes/herbicide-application/herbicide-application.component.ts]: You mean, `Comments`?

File: app/lucy/src/app/directives/element-focus/element-focus.directive.ts (hunk 2 above)
Comment (fwpushan) [reply app/lucy/src/app/directives/element-focus/element-focus.directive.ts]: as general code comment

File: app/lucy/src/app/components/Routes/herbicide-application/herbicide-application.component.ts (hunk 3 above)
Comment (fwpushan) [reply app/lucy/src/app/components/Routes/herbicide-application/herbicide-application.component.ts]: Yes, as general code comment

**Review by fwpushan — APPROVED** [review]

**Pull request merged**